src/extension/content-script/onend.js

```diff
@@ -32,11 +32,15 @@ async function init() {
 
   injectScript(browser.runtime.getURL("js/inpageScript.bundle.js")); // registers the DOM event listeners and checks webln again (which is also loaded onstart
 
-  // extract LN data from websites
   browser.runtime.onMessage.addListener((request, sender, sendResponse) => {
+    // extract LN data from websites
     if (request.action === "extractLightningData") {
       extractLightningData();
     }
+    // forward account switched messages to the website
+    else if (request.action === "accountSwitched") {
+      window.postMessage("accountSwitched", "*");
```

Comment:
how should the user use this? 

as a developer we want to do something like?: 
```
winow.addEventListener("webln:accountChanged", (old, new) => { ... }); 
```
is this correct?

Reply:
Just for reference: 

https://docs.metamask.io/wallet/reference/provider-api/#accountschanged

Seems they offer an `on()` method directly on the provider.

Reply:
@bumi  I added an example above but it requires listening to generic window messages. I like the solution @reneaaron linked because we can add the method to the existing `webln` and `nostr` objects, messages need to be explicitly subscribed to by the web app (and we can control whether or not the user needs to approve the request) and also have better control over the response object.

For the specific "accountSwitched" case I think we should be able to execute the method without asking for the user's permission because this is yet another modal users have to accept (We could consider having a different message type if we want to provide sensitive details in the callback?).

Should we go with a similar method as metamask?

`window.webln.on("accountSwitched", callback) `
`window.nostr.on("accountSwitched", callback) `

I think Alby should have a working implementation before we submit a PR for the specs.

Thank you both for bring these points up! much better than my original implementation, but I am glad the ball is rolling now :-)

Reply:
@reneaaron @bumi any thoughts on the event name? I called it "switched" here as "changed" could be misinterpreted as "updated". What do you think?

Reply:
To me `accountChanged` feels more natural as I think there is no reason for a website to be notified when something IN the account changes, right? It's maybe also more in line with other HTML events that offer onChange event handlers?

Reply:
CC @pavanjoshi914 - I think you preferred `accountChanged` too. Should we update it?

Reply:
sure!